File: common/src/main/java/com/ssblur/alchimiae/screen/screen/AlchimiaeScreens.java
package com.ssblur.alchimiae.screen.screen;

import com.ssblur.alchimiae.screen.menu.AlchimiaeMenus;
import dev.architectury.registry.menu.MenuRegistry;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public class AlchimiaeScreens {
  public static void register() {
    AlchimiaeMenus.BOILER.listen(type -> MenuRegistry.registerScreenFactory(type, BoilerScreen::new));
  }
}

File: common/src/main/kotlin/com/ssblur/alchimiae/screen/screen/AlchimiaeScreens.kt
package com.ssblur.alchimiae.screen.screen

import com.ssblur.alchimiae.AlchimiaeMod
import com.ssblur.alchimiae.screen.menu.AlchimiaeMenus
import com.ssblur.unfocused.event.client.ClientScreenRegistrationEvent.registerScreen
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment

@Environment(EnvType.CLIENT)
object AlchimiaeScreens {
  fun register() {
    AlchimiaeMenus.BOILER.then {
      AlchimiaeMod.registerScreen(it) { abstractContainerMenu, inventory, component ->
        BoilerScreen(abstractContainerMenu, inventory, component)
      }
    }

    AlchimiaeMenus.ALEMBIC.then {
      AlchimiaeMod.registerScreen(it) { abstractContainerMenu, inventory, component ->
        AlembicScreen(abstractContainerMenu, inventory, component)
      }
    }

    AlchimiaeMenus.FILTER.then {
      AlchimiaeMod.registerScreen(it) { abstractContainerMenu, inventory, component ->
        FilterScreen(abstractContainerMenu, inventory, component)
      }
    }

    AlchimiaeMenus.ALCHEMINDEX.then {
      AlchimiaeMod.registerScreen(it) { abstractContainerMenu, inventory, component ->
        AlchemindexScreen(abstractContainerMenu, inventory, component)
      }
    }
  }
}
